doc/shield-coinbase.md

```diff
@@ -0,0 +1,101 @@
+
+    z_shieldcoinbase fromaddress toaddress (fee) (limit)
+
+Where the default fee is 0.0010000 ZEC and the limit is 50.
+
+## Examples
+
+Sweep up coinbase utxos from a transparent address you use for mining:
+
+    zcash-cli z_shieldcoinbase tMyMiningAddress zMyPrivateAddress
+
+Sweep up coinbase utxos from multiple transparent addresses to a shielded address:
+
+    zcash-cli z_shieldcoinbase "*" zMyPrivateAddress
+
+Sweep up with a fee of 1.23 ZEC:
+
+    zcash-cli z_shieldcoinbase tMyMiningAddress zMyPrivateAddress 1.23
+
+Sweep up with a fee of 0.1 ZEC and set limit on the maximum number of utxos to shield at 25:
+
+    zcash-cli z_shieldcoinbase "*" zMyPrivateAddress 0.1 25
+
+### Asynchronous Call
+
+The `z_shieldcoinbase` RPC call is an asynchronous call, so you can queue up multiple operations. 
+
+When you invoke
+
+    zcash-cli z_shieldcoinbase tMyMiningAddress zMyPrivateAddress
+
+JSON will be returned immediately, with the following data fields populated:
+
+- operationid: a temporary id to use with `z_getoperationstatus` and `z_getoperationresult` to get the status and result of the operation.
+- shieldedUTXOs: number of coinbase utxos being shielded
+- shieldedValue: value of coinbase utxos being shielded.
+- remainingUTXOs: number of coinbase utxos still available for shielding.
+- remainingValue: value of coinbase utxos still available for shielding
+
+### Locking UTXOs
+
+The `z_shieldcoinbase` call will lock any selected UTXOs. This prevents the selected UTXOs which are already queued up from being selected for any other send operation.  If the `z_shieldcoinbase` call fails, any locked UTXOs are unlocked.
+
+You can use the RPC call `lockunspent` to see which UTXOs have been locked.  You can also use this call to unlock any UTXOs in the event of an unexpected system failure which leaves UTXOs in a locked state.
+
+### Limits, Performance and Transaction Confirmation
+
+The number of coinbase utxos selected for shielding can be adjusted by setting the limit parameter. The default value is 50.
+
+If the limit parameter is set to zero, the zcashd `mempooltxinputlimit` option will be used instead, where the default value for `mempooltxinputlimit` is zero which means no limit.
+
+Any limit is constrained by a hard limit due to the consensus rule defining a maximum transaction size of 100,000 bytes.
+
+In general, the more utxos that are selected, the longer it takes for the transaction to be verified.  Due to the quadratic hashing problem, some miners use the `mempooltxinputlimit` option to reject transactions with a large number of UTXO inputs.
+
+Currently, as of November 2017, there is no commonly agreed upon limit, but as a rule of thumb (a form of emergent consensus) if a transacion has less than 100 UTXO inputs, the transaction will be mined promptly by the majority of mining pools, but if it has say 500 UTXO inputs it might take several days to be mined by a miner who has higher or no limits.
```

Comment:
"if it has say 500 UTXO inputs" --> "if it has many more UTXO inputs, such as 500"

Reply:
ok